pkg/models/plan_general_characteristics.go

```diff
@@ -0,0 +1,138 @@
+
+	// Page 5
+	RulemakingRequired            *bool          `json:"rulemakingRequired" db:"rulemaking_required"`
+	RulemakingRequiredDescription *string        `json:"rulemakingRequiredDescription" db:"rulemaking_required_description"`
+	RulemakingRequiredNote        *string        `json:"rulemakingRequiredNote" db:"rulemaking_required_note"`
+	AuthorityAllowances           pq.StringArray `json:"authorityAllowances" db:"authority_allowances"`
+	AuthorityAllowancesOther      *string        `json:"authorityAllowancesOther" db:"authority_allowances_other"`
+	AuthorityAllowancesNote       *string        `json:"authorityAllowancesNote" db:"authority_allowances_note"`
+	WaiversRequired               *bool          `json:"waiversRequired" db:"waivers_required"`
+	WaiversRequiredTypes          pq.StringArray `json:"waiversRequiredTypes" db:"waivers_required_types"`
+	WaiversRequiredNote           *string        `json:"waiversRequiredNote" db:"waivers_required_note"`
+
+	// Meta
+	CreatedBy   *string    `json:"createdBy" db:"created_by"`
+	CreatedDts  *time.Time `json:"createdDts" db:"created_dts"`
+	ModifiedBy  *string    `json:"modifiedBy" db:"modified_by"`
+	ModifiedDts *time.Time `json:"modifiedDts" db:"modified_dts"`
+	Status      TaskStatus `json:"status" db:"status"`
+}
+
+// CalcStatus returns a TaskStatus based on how many fields have been entered in the PlanGeneralCharacteristics struct
+func (gc *PlanGeneralCharacteristics) CalcStatus() {
+
+	// TODO implement this. Doing it like this for now :)
+	gc.Status = TaskReady
+
+	//TODO look into making a generic function that takes in any parent class object and calcs status
+	// fieldCount := 5
+	// filledField := 0
+	// decidedStat := TaskReady
+
+	// if gc.ModelType != nil {
+	// 	filledField++
+	// }
+
+	// if gc.Problem != nil {
+	// 	filledField++
+	// }
+	// if gc.Goal != nil {
+	// 	filledField++
+	// }
+	// if gc.TestInventions != nil {
+	// 	filledField++
+	// }
+	// if gc.Note != nil {
+	// 	filledField++
+	// }
+
+	// if filledField == fieldCount {
+	// 	decidedStat = TaskComplete
+
+	// } else if filledField > 0 {
+	// 	decidedStat = TaskInProgress
+	// }
+	// gc.Status = decidedStat
+}
+
+// GetModelTypeName returns a string name that represents the PlanBasics struct
+func (gc PlanGeneralCharacteristics) GetModelTypeName() string {
+	return "Plan_General_Characteristics"
+}
+
+// GetID returns the ID property for a PlanBasics struct
+func (gc PlanGeneralCharacteristics) GetID() uuid.UUID {
+	return gc.ID
+}
+
+// GetPlanID returns the ModelPlanID property for a PlanBasics struct
+func (gc PlanGeneralCharacteristics) GetPlanID() uuid.UUID {
+	return gc.ModelPlanID
+}
+
+// GetModifiedBy returns the ModifiedBy property for a PlanBasics struct
+func (gc PlanGeneralCharacteristics) GetModifiedBy() *string {
+	return gc.ModifiedBy
+}
```

Comment:
After merging with the nullable modifiedby PR, you will need to modify this method a bit (dereference) and implement PlanGeneralCharacteristics.CreatedBy